runtime: python
env: flex
# --threads 48 fails with ERROR: (gcloud.app.deploy) Error Response: [13] An internal error occurred during deployment. You may need to delete this version manually.
entrypoint: gunicorn --timeout 3600 --workers 8 --threads 24 -b :$PORT fit:app

runtime_config:
  python_version: 2

service: dev
#service: default

env_variables:
  CLIENT_SECRET: 'secrets_do_not_git_push/client_secret.apps.googleusercontent.com.json'
  APP_CONFIG: 'app_secret.config'

beta_settings:
  cloud_sql_instances: [Instance connection name]

# The settings below are to reduce costs during testing and are not appropriate
# for production use. For more information, see:
# https://cloud.google.com/appengine/docs/flexible/python/configuring-your-app-with-app-yaml
manual_scaling:
  instances: 1
resources:
  cpu: 1
  memory_gb: 0.5
  disk_size_gb: 10

# default production App Engine service recommended scaling
#automatic_scaling:
#  min_num_instances: 1
#  max_num_instances: 10
#  cpu_utilization:
#    target_utilization: 0.6